### ISSUE TITLE
Duplicate Icons in TaskBar

### ISSUE THREAD

**Issue by njmulsqb** [linuxdeepin/developer-center#2352]

**Comment by tsic404** [comment]
look like this? ![image](https://user-images.githubusercontent.com/48748112/159209855-0ca5dc51-f402-4444-8e18-3bf44b06a62f.png)

**Comment by njmulsqb** [comment]
> look like this? ![image](https://user-images.githubusercontent.com/48748112/159209855-0ca5dc51-f402-4444-8e18-3bf44b06a62f.png)
> 

Yes, this one is close. I removed deepin long ago so cant help much with screenshots

**Comment by NjlsShade** [comment]
I also am seeing this issue, however, it's most commonly with Flatpaks. Not sure if it can be easily resolved without breaking other things?
![Screenshot_select-area_20220521001840](https://user-images.githubusercontent.com/15788214/169635083-856938e8-0799-4a89-af7f-4c25afab0109.png)

I opened Unity Hub using the second to last icon on the right, then another one appeared to its right (last two icons on the right are the same app).

**Comment by tsic404** [comment]
> I also am seeing this issue, however, it's most commonly with Flatpaks. Not sure if it can be easily resolved without breaking other things? ![Screenshot_select-area_20220521001840](https://user-images.githubusercontent.com/15788214/169635083-856938e8-0799-4a89-af7f-4c25afab0109.png)
> 
> I opened Unity Hub using the second to last icon on the right, then another one appeared to its right (last two icons on the right are the same app).

This should be different, you get a WM_CLASS that is different from the one in the desktop file, In the screenshot above, two icons are mapped to one opened applications, one closes the other closed too.

**Comment by tsic404** [comment]
> I also am seeing this issue, however, it's most commonly with Flatpaks. Not sure if it can be easily resolved without breaking other things? ![Screenshot_select-area_20220521001840](https://user-images.githubusercontent.com/15788214/169635083-856938e8-0799-4a89-af7f-4c25afab0109.png)
> 
> I opened Unity Hub using the second to last icon on the right, then another one appeared to its right (last two icons on the right are the same app).

Your docked app is `unity hub` opened is `unity editor`?
